Shopify/eng-intern-assessment-rails
# Shopify Backend Engineering Internship Assessment

## Demo
https://www.loom.com/share/c3f77b0883a942e99e0eab4dc5cce892

## Implementation description
- Wrote CRUD implementations for Article controller
- Article model changes for db communication
- Updated Article views with UI elements to support CRUD functionality
- Added search and table sorting features
- Added additional model tests and created controller tests
- Added dummy data for seed

## Steps to test
3. `bundle install`
4. `rails db:migrate`
5. `rails db:seed` 
6. Run `rails test:all` to confirm all test cases pass
7. Perform user actions for CRUD, search, and filtering using UI to test expected behaviour
8. Ensure code quality and best practices for Ruby on Rails project



## Pictures

All articles:
<img width="1502" alt="image" src="https://github.com/Shopify/eng-intern-assessment-rails/assets/58372785/887ae0b7-2403-482c-82f7-e3c9dc8f8741">

Create an article:
<img width="1512" alt="image" src="https://github.com/Shopify/eng-intern-assessment-rails/assets/58372785/d207a5e5-5a02-4c25-a997-7b41350aab39">

Read an article:
<img width="1508" alt="image" src="https://github.com/Shopify/eng-intern-assessment-rails/assets/58372785/5146c3ce-f163-4a7b-8e7b-012cb37b4af5">


Update/Edit an article:
<img width="1499" alt="image" src="https://github.com/Shopify/eng-intern-assessment-rails/assets/58372785/e20b49f6-57de-4820-a541-1b95c9e4de15">


Delete an article:

Deleted lots of articles compared to above:
<img width="1507" alt="image" src="https://github.com/Shopify/eng-intern-assessment-rails/assets/58372785/cd1c2e7b-c49f-44a0-8546-201a83671287">

Search:
<img width="1509" alt="image" src="https://github.com/Shopify/eng-intern-assessment-rails/assets/58372785/0ab012c2-a4b4-41af-8ac5-65de62f42f23">


Sorting by date (order columns are possible as well):
<img width="1501" alt="image" src="https://github.com/Shopify/eng-intern-assessment-rails/assets/58372785/6b0db694-52ec-4497-bce7-00df6eafa549">
